### DISCUSSION TITLE
python_read_binary之疑惑😅

### DISCUSSION THREAD

**Discussion by JunTingLin** [1101-GS4719-C/Python-Community-GS4719-C#589] [Q&A]
# python_read_binary之疑惑😅
## 前言
上週老師上課教到檔案的讀寫，可以寫入二進位也可以讀出二進位。
我在mod18/note.txt，轉utf-8編碼，並打上以下字串
```
你是我的眼
帶我領略四季的變化
```

## 先執行rt (read text)，並用utf-8編碼
![image](https://user-images.githubusercontent.com/92431095/167253952-de9fb56b-b108-4c2e-b6de-8cd8e982f639.png)

完全符合預期

## 此時將模式改成rb (read binary)
![image](https://user-images.githubusercontent.com/92431095/167253966-852a4a4f-43ce-4b84-a541-3e686dca3d3c.png)

### 問題
1. 出來的好像是16進位??
2. 出來的16進位好像也跟utf-8編碼對不上??

一般凡人看得懂的中文
```你是我的眼```

這是python跑出的結果(\r迴車\n換行)
```\xe4\xbd\xa0\xe6\x98\xaf\xe6\x88\x91\xe7\x9a\x84\xe7\x9c\xbc\r\n```
5個中文字，15個bytes，一個中文字3 bytes

這是中文轉utf-8的
```\u4f60\u662f\u6211\u7684\u773c```
5個中文字，10bytes，一個中文字2 bytes
![image](https://user-images.githubusercontent.com/92431095/167253978-e4c6abe2-f5af-4466-9f07-b46da9a3f38d.png)

**Comment by JSHT** [comment]
utf-8 有其編碼格式，他可能一個 unicode 符號就佔 3bytes 甚至更多
找到一篇講得蠻詳細的 [UTF-8编码占几个字节?](https://blog.csdn.net/bluetjs/article/details/52936943)

**Comment by Ku-Hao-Chen** [comment]
謝謝分享，原來還有這種方式。

**Comment by Sazmafuji** [comment]
發現此事，好厲害
